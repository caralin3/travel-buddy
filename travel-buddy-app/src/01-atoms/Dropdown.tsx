import React from 'react';
import { Dropdown as RBDropdown, DropdownItem, DropdownMenu, DropdownToggle } from 'reactstrap';

export interface DropdownMenuItem {
  label: string;
  onClick: VoidFunction;
}

export interface DropdownProps {
  isOpen?: boolean;
  label: string;
  menuItems: DropdownMenuItem[];
  toggle: VoidFunction;
}

export const Dropdown: React.FC<DropdownProps> = ({ isOpen = false, label, menuItems, toggle }) => {
  return (
    <RBDropdown isOpen={isOpen} toggle={toggle}>
      <DropdownToggle caret>{label}</DropdownToggle>
      <DropdownMenu end>
        {menuItems.map((menuItem) => (
          <DropdownItem key={menuItem.label} onClick={menuItem.onClick}>
            {menuItem.label}
          </DropdownItem>
        ))}
      </DropdownMenu>
    </RBDropdown>
  );
};
