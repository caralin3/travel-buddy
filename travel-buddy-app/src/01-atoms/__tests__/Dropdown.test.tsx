import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Dropdown } from '../Dropdown';

describe('Dropdown', () => {
  const mockToggle = jest.fn();

  const wrapper = (isOpen?: boolean) => {
    const props = {
      isOpen,
      label: 'Dropdown',
      menuItems: [
        {
          label: 'One',
          onClick: jest.fn(),
        },
        { label: 'Two', onClick: jest.fn() },
      ],
      toggle: mockToggle,
    };

    return shallow(<Dropdown {...props} />);
  };

  it('should render correctly', () => {
    const component = wrapper();
    expect(toJson(component)).toMatchSnapshot();
  });
  it('should render open correctly', () => {
    const component = wrapper(true);
    expect(toJson(component)).toMatchSnapshot();
  });
});
