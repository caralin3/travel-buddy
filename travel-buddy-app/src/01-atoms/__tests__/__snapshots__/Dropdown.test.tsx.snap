// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should render correctly 1`] = `
<Dropdown
  a11y={true}
  active={false}
  direction="down"
  inNavbar={false}
  isOpen={false}
  nav={false}
  setActiveFromChild={false}
  toggle={[MockFunction]}
>
  <DropdownToggle
    aria-haspopup={true}
    caret={true}
    color="secondary"
  >
    Dropdown
  </DropdownToggle>
  <DropdownMenu
    end={true}
    flip={true}
    modifiers={Array []}
    tag="div"
  >
    <DropdownItem
      key="One"
      onClick={[MockFunction]}
      tag="button"
      toggle={true}
    >
      One
    </DropdownItem>
    <DropdownItem
      key="Two"
      onClick={[MockFunction]}
      tag="button"
      toggle={true}
    >
      Two
    </DropdownItem>
  </DropdownMenu>
</Dropdown>
`;

exports[`Dropdown should render open correctly 1`] = `
<Dropdown
  a11y={true}
  active={false}
  direction="down"
  inNavbar={false}
  isOpen={true}
  nav={false}
  setActiveFromChild={false}
  toggle={[MockFunction]}
>
  <DropdownToggle
    aria-haspopup={true}
    caret={true}
    color="secondary"
  >
    Dropdown
  </DropdownToggle>
  <DropdownMenu
    end={true}
    flip={true}
    modifiers={Array []}
    tag="div"
  >
    <DropdownItem
      key="One"
      onClick={[MockFunction]}
      tag="button"
      toggle={true}
    >
      One
    </DropdownItem>
    <DropdownItem
      key="Two"
      onClick={[MockFunction]}
      tag="button"
      toggle={true}
    >
      Two
    </DropdownItem>
  </DropdownMenu>
</Dropdown>
`;
